typescript/packages/wallets/sui/src/SuiKeyPairWalletClient.ts

```diff
@@ -0,0 +1,59 @@
+import type { Signature } from "@goat-sdk/core";
+import { SuiClient, type SuiTransactionBlockResponse } from "@mysten/sui.js/client";
+import { type Ed25519Keypair } from "@mysten/sui.js/keypairs/ed25519";
+import { TransactionBlock } from "@mysten/sui.js/transactions";
+import { SuiWalletClient } from "./SuiWalletClient";
+import type { SuiQuery, SuiReadResponse, SuiTransaction, Transaction } from "./types";
+
+export type SuiKeyPairWalletClientParams = {
+    client: SuiClient;
+    keypair: Ed25519Keypair;
+};
+
+export class SuiKeyPairWalletClient extends SuiWalletClient {
+    private keypair: Ed25519Keypair;
+
+    constructor(params: SuiKeyPairWalletClientParams) {
+        super({ client: params.client });
+        this.keypair = params.keypair;
+    }
+
+    async sendTransaction(transaction: SuiTransaction) {
```

Comment:
No, check the docs here: https://sdk.mystenlabs.com/typescript/transaction-building/basics

You use the transaction object, not transactionBlock

Also you need to wait for the tx:

`await client.waitForTransaction({ digest: result.digest });`